x/ecocredit/server/marketplace/sell.go

```diff
@@ -92,3 +100,62 @@ func (k Keeper) getOrCreateMarketId(ctx context.Context, creditTypeAbbrev, bankD
 		return 0, err
 	}
 }
+
+func (k Keeper) escrowCredits(ctx context.Context, ownerAcc sdk.AccAddress, batchId uint64, sellQty math.Dec) error {
+	// get the seller's balance
+	bal, err := k.coreStore.BatchBalanceStore().Get(ctx, ownerAcc, batchId)
+	if err != nil {
+		return err
+	}
+	tradable, err := math.NewDecFromString(bal.Tradable)
+	if err != nil {
+		return err
+	}
+	// subtract the desired sell amount from tradable balance
+	newTradable, err := math.SafeSubBalance(tradable, sellQty)
+	if err != nil {
+		return fmt.Errorf("tradable balance: %s, sell order request: %s - %w", tradable.String(), sellQty.String(), err)
+	}
+
+	escrowed, err := math.NewDecFromString(bal.Escrowed)
+	if err != nil {
+		return err
+	}
+	newEscrowed, err := math.SafeAddBalance(escrowed, sellQty)
+	if err != nil {
+		return fmt.Errorf("escrowed balance: %s, sell order request: %s - %w", escrowed.String(), sellQty.String(), err)
+	}
+	// set the new balance
+	bal.Tradable = newTradable.String()
+	bal.Escrowed = newEscrowed.String()
+
+	// save
+	if err = k.coreStore.BatchBalanceStore().Update(ctx, bal); err != nil {
+		return err
+	}
+
+	// update the batch supply
+	supply, err := k.coreStore.BatchSupplyStore().Get(ctx, batchId)
+	if err != nil {
+		return err
+	}
+	supTradable, err := math.NewDecFromString(supply.TradableAmount)
+	if err != nil {
+		return err
+	}
+	supEscrow, err := math.NewDecFromString(supply.EscrowedAmount)
+	if err != nil {
+		return err
+	}
+	supTradable, err = math.SafeSubBalance(supTradable, sellQty)
+	if err != nil {
+		return err
+	}
+	supEscrow, err = math.SafeAddBalance(supEscrow, sellQty)
+	if err != nil {
+		return err
+	}
+	supply.EscrowedAmount = supEscrow.String()
+	supply.TradableAmount = supTradable.String()
```

Comment:
@aaronc shouldn't we have another type to serialize big numbers in the state? One type for response, another type for state.

Reply:
I don't think so. State is directly exposed to clients and in the absence of any standard that clients can decode, string is the best choice. If someone defines a standard that's well specified this could change in the future